lib/babel-options-util.js

```diff
@@ -406,16 +407,23 @@ function _shouldIncludeDecoratorPlugins(config) {
 
 function _shouldHandleTypeScript(config, parent) {
   let emberCLIBabelConfig = config["ember-cli-babel"] || {};
+
   if (typeof emberCLIBabelConfig.enableTypeScriptTransform === "boolean") {
     return emberCLIBabelConfig.enableTypeScriptTransform;
   }
-  let typeScriptAddon =
-    parent.addons &&
-    parent.addons.find((a) => a.name === "ember-cli-typescript");
-  return (
-    typeof typeScriptAddon !== "undefined" &&
-    semver.gte(typeScriptAddon.pkg.version, "4.0.0-alpha.1")
-  );
+
+  let tsDependency =
+    parent.pkg &&
+    parent.pkg.dependencies &&
+    parent.pkg.dependencies["ember-cli-typescript"];
+
+  if (typeof tsDependency !== "undefined") {
```

Comment:
We can probably just `tsDependency !== undefined` (no need for typeof check here)